lib/blocks.php

```diff
@@ -528,3 +528,66 @@ function _gutenberg_footnotes_force_filtered_html_on_import_filter( $arg ) {
 add_action( 'init', '_gutenberg_footnotes_kses_init' );
 add_action( 'set_current_user', '_gutenberg_footnotes_kses_init' );
 add_filter( 'force_filtered_html_on_import', '_gutenberg_footnotes_force_filtered_html_on_import_filter', 999 );
+
+/**
+ * Filters the arguments for registering a post type adding the footnotes support by default.
+ *
+ * @access private
+ *
+ * @param array  $args      Array of arguments for registering a post type.
+ *                          See the register_post_type() function for accepted arguments.
+ */
+function _gutenberg_add_footnotes_post_type_support( $args ) {
+	if ( ! empty( $args['supports'] ) ) {
+		$supports = &$args['supports'];
+		if ( in_array( 'editor', $supports, true ) && in_array( 'custom-fields', $supports, true ) && in_array( 'revisions', $supports, true ) && ! in_array( 'footnotes', $supports, true ) ) {
+			$supports[] = 'footnotes';
+		}
+	}
+
+	return $args;
+}
+
+/**
+ * Adds the footnotes support to the post type by default.
+ */
+add_filter(
+	'register_post_type_args',
+	'_gutenberg_add_footnotes_post_type_support'
+);
+
+/**
+ * Registers the footnotes meta field for post types that support it.
+ *
+ * @internal
+ *
+ */
+function _gutenberg_register_footnotes_meta_field() {
+	$post_types = get_post_types(
+		array(
+			'show_in_rest' => true,
+			'public'       => true,
+		)
+	);
+	foreach ( $post_types as $post_type ) {
+		// Only register the meta field if the post type supports the editor, custom fields, and revisions.
+		if ( post_type_supports( $post_type, 'editor' ) && post_type_supports( $post_type, 'custom-fields' ) && post_type_supports( $post_type, 'revisions' ) && post_type_supports( $post_type, 'footnotes' ) ) {
+			$post_type_meta_keys = get_registered_meta_keys( 'post', $post_type );
+			if ( ! isset( $post_type_meta_keys['footnotes'] ) ) {
+				register_post_meta(
+					$post_type,
+					'footnotes',
+					array(
+						'show_in_rest'      => true,
+						'single'            => true,
+						'type'              => 'string',
+						'revisions_enabled' => true,
+					)
+				);
+			}
+		}
+	}
+}
+
+// Registers the footnotes meta field for post types that support it.
+add_action( 'init', '_gutenberg_register_footnotes_meta_field', 100 );
```

Comment:
A note explaining this runs late to allow for plugins registering post types would be helpful here.

@spacedmonkey @TimothyBJacobs does `init, 100` run in the REST API or does `rest_api_init` exit beforehand on `init, 10`?